sql/wsrep_sst.cc

```diff
@@ -89,18 +89,21 @@ static Regex sst_allowed_methods_regex;
    characters and the ways they could be used to inject the command to
    the OS. However this approach seems to be too error prone.
    Instead of this we will just allow alpha-num + a few special characters
-   (colon, slash, dot, underscore, square brackets, hyphen).
+   (colon, slash, dot, underscore, square brackets, hyphen, '@' symbol).
 
-   It is the same as ^[\w:/.[\]-]+$    */
+   Note: '@' symbol is possible with encrypted rsync SST, where data can be
+         "localhost:0141b8fc9fa853ac9d1d48a383de2382@127.0.0.1:13009/rsync_sst"
+
+   It is the same as ^[\w:/.[\]-@]+$    */
 
    /* For some reason regex engine is not able to handle the following pattern
    allowing square brackets:
-   "^[[:alnum:]:/._[\\]-]+$"
+   "^[[:alnum:]:/._[\\]-@]+$"
    We will substitute square brackets with lt/gt for validation.
    Because of another regex engine problem, hyphen literal has to be located
    at the end of matching set (cannot be escaped when in the middle). */
 static const char *sst_method_allowed_chars_regex_pattern=
-    "^[[:alnum:]:/._<>-]+$";
+    "^[[:alnum:]:/._<>-@]+$";
```

Comment:
Did you consider the problem with the hyphen mentioned in the comment? Right now it will be considered as the range between > and @, no?

Reply:
Yes. You're right! I remember this conversation. Thanks for your inputs. I'll modify it.

Reply:
Done